platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/schedulers/SequentialThreadTaskScheduler.java

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.wiring.schedulers;
+
+import static java.util.concurrent.TimeUnit.NANOSECONDS;
+
+import com.swirlds.base.state.Startable;
+import com.swirlds.base.state.Stoppable;
+import com.swirlds.common.metrics.extensions.FractionalTimer;
+import com.swirlds.common.wiring.TaskScheduler;
+import com.swirlds.common.wiring.WiringModel;
+import com.swirlds.common.wiring.builders.TaskSchedulerType;
+import com.swirlds.common.wiring.counters.ObjectCounter;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.lang.Thread.UncaughtExceptionHandler;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Consumer;
+
+/**
+ * A scheduler that performs work sequentially on a dedicated thread. This class has very similar semantics to
+ * {@link DirectTaskScheduler}, except that work is done on a thread instead of on a fork join pool.
+ *
+ * @param <OUT>
+ */
+public class SequentialThreadTaskScheduler<OUT> extends TaskScheduler<OUT> implements Startable, Stoppable {
+
+    private final UncaughtExceptionHandler uncaughtExceptionHandler;
+    private final ObjectCounter onRamp;
+    private final ObjectCounter offRamp;
+    private final FractionalTimer busyTimer;
+    private final Duration sleepDuration;
+
+    private final BlockingQueue<SequentialThreadTask> tasks = new LinkedBlockingQueue<>();
+
+    private static final int BUFFER_SIZE = 1024;
```

Comment:
I think we need this to be configurable

Reply:
Configuring the size of this buffer used to be important with the old `QueueThread` implementation, but it's not important here.

With the `QueueThread` implementation, items sitting in the buffer would not count towards the maximum allowable capacity, since we were enforcing capacity using the queue. In all scenarios of which I am aware, the only reason we would ever set a buffer size that was not the default was when we intentionally wanted a very small capacity, and did not want the buffer to significantly inflate that capacity.

In the new wiring implementation does count the items in the buffer against the available capacity. Unless I'm missing something, I don't think we have any use cases that would really care about this capacity, as long as a sane value is chosen.